redis/src/cluster_async/mod.rs

```diff
@@ -1108,15 +1139,42 @@ where
         }
     }
 
+    /// This function takes a node's address, examines if its host has encountered a DNS change, where the node's endpoint now leads to a different IP address.
+    /// If no socket addresses are discovered for the node's host address, or if it's a non-DNS address, it returns false.
+    /// In case the node's host address resolves to socket addresses and none of them match the current connection's IP, a DNS change is detected, resulting in a true return.
+    async fn is_dns_changed(addr: &str, curr_ip: &String) -> bool {
+        let (host, port) = match get_host_and_port_from_addr(addr) {
+            Some((host, port)) => (host, port),
+            None => return false,
+        };
+        let updated_addresses = match get_socket_addrs(host, port).await {
+            Ok(socket_addrs) => socket_addrs,
+            Err(_) => return false,
+        };
+        for socket_addr in updated_addresses {
```

Comment:
use `contains` instead

Reply:
updated_addresses is an iterator, to use contains will require to do collect it to a vector, isn't it?